model/websocket_client.go

```diff
@@ -249,10 +247,15 @@ func (wsc *WebSocketClient) SendMessage(action string, data map[string]interface
 	req.Data = data
 
 	wsc.Sequence++
-	wsc.writeChan <- writeMessage{
+	// we do a non-blocking send in case the writer has already exited.
+	select {
+	case wsc.writeChan <- writeMessage{
```

Comment:
Should we make `writeChan` buffered? My worry with this is that we might end up dropping messages during normal operations if the `writer` goroutine takes a bit longer to handle the message.